### PR TITLE
[BEAM-239]: Rename RemoveDuplicates to Distinct

### PR DESCRIPTION
**Pull request by nssalian**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:
- [x] Done in the PR:
   Refactored the project to replace RemoveDuplicates to Distinct
- [x] Ran Travis Locally and mvn clean install with SUCCESS to verify
- [x] Replace `<Jira issue #>` in the title with the actual Jira issue
     number, if there is one.
- [ ] If this contribution is large, please file an Apache
     [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).
---

Requesting review.

### CODE REVIEW

**Comment by robertwb** [comment]
I think we should stick with verbs for PTransforms, but whatever we do we need to change Python as well.

**Comment by nssalian** [comment]
@robertwb ,thanks for the response. If it is ok, I can perhaps open a new PR/ JIRA for the Python bit after this one goes in.

**Comment by nssalian** [comment]
@eljefe6a do you have some time to help review this?

**Comment by dhalperi** [comment]
I think Robert had already volunteered as a reviewer :).

R: @robertwb 

I'm thinking harder about Robert's comment. It does seem like most transforms are explicitly verbs, though I do not think this has ever been documented.

And there are plenty of exceptions, like `Values`.

Short summary: I'm okay with this change, but it may just be better to update the Javadoc. Either way, may be worth discussing on the Beam list.

**Comment by eljefe6a** [comment]
Either way, the JavaDoc for Distinct/RemoveDuplicates needs to be updated. It should talk about taking the `PCollection` and creating a distinct list of elements in it.

**Comment by nssalian** [comment]
@dhalperi ,  let me start a topic on it. Thank you. :)

**Comment by nssalian** [comment]
@eljefe6a  thank you.I'll add it to the javadoc in my next commit.

**Comment by nssalian** [comment]
Changing it to MakeDistinct as per the discussion on dev@. Will post a commit soon.

**Comment by nssalian** [comment]
@eljefe6a , @robertwb , @dhalperi , any thoughts after the discussion?

File: examples/java/src/main/java/org/apache/beam/examples/complete/TfIdf.java [hunk 1]
```diff
@@ -209,7 +209,7 @@ public ComputeTfIdf() { }
       final PCollectionView<Long> totalDocuments =
           uriToContent
           .apply("GetURIs", Keys.<URI>create())
-          .apply("RemoveDuplicateDocs", RemoveDuplicates.<URI>create())
+          .apply("RemoveDuplicateDocs", Distinct.<URI>create())
```
Comment (dhalperi) [inline examples/java/src/main/java/org/apache/beam/examples/complete/TfIdf.java]: (driveby: maybe update the names of the steps, if you think there is a clearer name)

Comment (nssalian) [inline examples/java/src/main/java/org/apache/beam/examples/complete/TfIdf.java]: Thanks @eljefe6a. I do see RemoveDuplicateTimestamps as well. Best to have that changed as well.
Posting a commit.

**Comment by nssalian** [comment]
Need to rebase. Will correct this.

**Comment by nssalian** [comment]
@dhalperi , @eljefe6a, let me know if I need to alter anything.

File: examples/java/src/test/java/org/apache/beam/examples/cookbook/DeDupExampleTest.java [hunk 2]
```diff
@@ -25,7 +25,7 @@
 import org.apache.beam.sdk.testing.RunnableOnService;
 import org.apache.beam.sdk.testing.TestPipeline;
 import org.apache.beam.sdk.transforms.Create;
-import org.apache.beam.sdk.transforms.RemoveDuplicates;
+import org.apache.beam.sdk.transforms.Distinct;
```
Comment (eljefe6a) [inline examples/java/src/test/java/org/apache/beam/examples/cookbook/DeDupExampleTest.java]: The name of this Test is `DeDupExampleTest.java` should be renamed.

File: sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/DistinctTest.java [hunk 3]
```diff
@@ -40,10 +40,10 @@
  * Tests for RemovedDuplicates.
```
Comment (nssalian) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/DistinctTest.java]: Will do.

File: examples/java/src/test/java/org/apache/beam/examples/cookbook/DeDupExampleTest.java (hunk 2 above)
Comment (nssalian) [inline examples/java/src/test/java/org/apache/beam/examples/cookbook/DeDupExampleTest.java]: DistinctTest?

**Comment by nssalian** [comment]
@dhalperi , please let me know if you have additional comments?

**Comment by dhalperi** [comment]
It's fine as-is, but Jesse should merge when he thinks it's ready

**Comment by eljefe6a** [comment]
Once my credentials are ready, I'll merge.

File: runners/spark/src/test/java/org/apache/beam/runners/spark/translation/streaming/KafkaStreamingTest.java [hunk 4]
```diff
@@ -187,7 +187,7 @@ private static Properties defaultProducerProps() {
     Properties producerProps = new Properties();
     producerProps.putAll(EMBEDDED_KAFKA_CLUSTER.getProps());
     producerProps.put("acks", "1");
-    producerProps.put("bootstrap.servers", EMBEDDED_KAFKA_CLUSTER.getBrokerList());
+      producerProps.put("bootstrap.servers", EMBEDDED_KAFKA_CLUSTER.getBrokerList());
```
Comment (nssalian) [inline runners/spark/src/test/java/org/apache/beam/runners/spark/translation/streaming/KafkaStreamingTest.java]: Fixed it. For some reason it went to the previous line when I tried before. Worked now.

**Comment by nssalian** [comment]
retest this please

**Comment by eljefe6a** [comment]
Do a rebase and push. That will kick off another build.

**Comment by nssalian** [comment]
Thanks @robertwb @dhalperi @eljefe6a for the review.
